src/calendar/__tests__/calendar.test.tsx

```diff
@@ -263,6 +265,302 @@ describe('disabled date', () => {
+              return 'Disabled on Sundays';
+            }
+
+            return '';
+          }}
+        />
+      );
+      const wrapper = createWrapper(container).findCalendar()!;
+
+      fireEvent.mouseEnter(wrapper.findDateAt(1, 6).getElement());
+
+      expect(wrapper.findDateAt(1, 6).findDisabledReason()!.getElement()).toHaveTextContent('Disabled on Saturdays');
+
+      fireEvent.mouseLeave(wrapper.findDateAt(1, 6).getElement());
+
+      expect(wrapper.findDateAt(1, 6).findDisabledReason()).toBe(null);
+    });
+
+    test('has no aria-describedby by default', () => {
+      const { container } = render(<Calendar {...defaultProps} value="2022-01-03" />);
+      const wrapper = createWrapper(container).findCalendar()!;
+
+      expect(wrapper.findDateAt(1, 6).getElement()).not.toHaveAttribute('aria-describedby');
+    });
+
+    test('has no aria-describedby without disabledReason', () => {
+      const { container } = render(
+        <Calendar
+          {...defaultProps}
+          value="2022-01-03"
+          isDateEnabled={(date: Date) => date.getDay() !== 6 && date.getDay() !== 0}
+        />
+      );
+      const wrapper = createWrapper(container).findCalendar()!;
+
+      expect(wrapper.findDateAt(1, 6).getElement()).not.toHaveAttribute('aria-describedby');
+    });
+
+    test('has aria-describedby with disabledReason', () => {
+      const { container } = render(
+        <Calendar
+          {...defaultProps}
+          value="2022-01-03"
+          isDateEnabled={(date: Date) => date.getDay() !== 6 && date.getDay() !== 0}
+          dateDisabledReason={(date: Date) => {
+            if (date.getDay() === 6) {
+              return 'Disabled on Saturdays';
+            }
+
+            if (date.getDay() === 0) {
+              return 'Disabled on Sundays';
+            }
+
+            return '';
+          }}
+        />
+      );
+
+      const wrapper = createWrapper(container).findCalendar()!;
+
+      expect(wrapper.findDateAt(1, 6).getElement()).toHaveAttribute('aria-describedby');
+    });
+
+    test('has hidden element (linked to aria-describedby) with disabledReason', () => {
+      const { container } = render(
+        <Calendar
+          {...defaultProps}
+          value="2022-01-03"
+          isDateEnabled={(date: Date) => date.getDay() !== 6 && date.getDay() !== 0}
+          dateDisabledReason={(date: Date) => {
+            if (date.getDay() === 6) {
+              return 'Disabled on Saturdays';
+            }
+
+            if (date.getDay() === 0) {
+              return 'Disabled on Sundays';
+            }
+
+            return '';
+          }}
+        />
+      );
+
+      const wrapper = createWrapper(container).findCalendar()!;
+
+      expect(wrapper.findDateAt(1, 6).findDisabledReasonDescription()!.getElement()).toHaveTextContent(
+        'Disabled on Saturdays'
+      );
+    });
+
+    test('space keydown does not trigger onChange when disabled with reason', () => {
```

Comment:
This test case was introduced because disabled dates with reason are focusable, which is not a case for normal `disabled` states